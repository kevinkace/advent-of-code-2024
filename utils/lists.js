export function getListDiffs([listA, listB]) {
    const diffs = [];

    for (let index = 0; index < listA.length; index++) {
        diffs.push(Math.abs(listA[index] - listB[index]));
    }

    return diffs;
}

export function sortLists(lists) {
    return lists.map(list => list.sort((a, b) => a > b ? 1 : -1));
}

export function sumList(list) {
    return list.reduce((acc, cur) => acc + cur, 0);
}