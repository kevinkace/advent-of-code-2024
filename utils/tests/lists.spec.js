import { expect, test } from "vitest";

import { getListDiffs, sumList } from "../lists";

test("getListSums", () => {
    expect(getListDiffs([[2, 3, 4], [4, 1, 4]])).toMatchSnapshot();
});

test("sumList", () => {
    expect(sumList([3, 56, 2])).toMatchSnapshot();
});
