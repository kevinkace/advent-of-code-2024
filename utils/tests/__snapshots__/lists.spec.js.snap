// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getListSums 1`] = `
[
  2,
  2,
  0,
]
`;

exports[`sumList 1`] = `61`;
